PX4/PX4-Autopilot
# Offboard separate trajectory_setpoint topic

**Describe problem solved by this pull request**
Fixes #18306 

**Describe your solution**
This adds the offboard_setpoint_trajectory topic to be used exclusively while in offboard mode.
The correct topic is decided if the controllers (consumer) `_vehicle_control_mode.flag_control_offboard_enabled` is true
This way when the autopilot is in an auto mode, flight_tasks trajectory_setpoints will not be overridden by offboard trajectory_setpoints. 

**Describe possible alternatives**
Let the offboard computer only publish trajectory_setpoint when nav_state is set to offboard

**Test data / coverage**
Tests in Gazebo
1. Send trajectory_setpoint at 100hz with offboard disabled in an auto mode and observe auto mode behaving correctly
2. Send offboard_trajectory_setpoint at 10hz with offboard enabled and observe offboard behaving correctly
